configs/next/packages/libnxz/stage_1

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Copyright 2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source ${AT_BASE}/scripts/utilities/bitsize_selection.sh
+
+# NX GZIP library build parameters for stage 1  64 bits
+# ======================================================================
+#
+
+# Tell the build system to hold the temp install folder
+ATCFG_HOLD_TEMP_INSTALL='no'
+# Tell the build system to hold the temp build folder
+ATCFG_HOLD_TEMP_BUILD='no'
+# Build in a new directory
+ATCFG_BUILD_STAGE_T='dir'
+
+atcfg_pre_configure() {
+	# The bin and lib directories depend on the word size, e.g.
+	# configure_bindir may be set to bin, bin32, or bin64.
+	configure_libdir=${configure_prefix}/$(find_build_libdir ${AT_BIT_SIZE})
+}
+
+atcfg_configure() {
+	# libnxz do not use autoconf
+	${SUB_MAKE} clean -C ${ATSRC_PACKAGE_WORK}
+	true
+}
+
+atcfg_make() {
+	# In gcc10 the default is now -fno-common. In order to not have multiple
+	# definitions -fcommon is needed.
+	configure_cc="${at_dest}/bin/${target64:-${target}}-gcc"
+	PATH="${at_dest}/bin:${PATH}" \
+	${SUB_MAKE} -C ${ATSRC_PACKAGE_WORK} \
+	CC="${configure_cc} -m${AT_BIT_SIZE}" \
+	FLG="-std=gnu11 -fcommon" ZLIB=""
```

Comment:
Why is this removing `ZLIB = -DZLIB_API` ?

Reply:
ZLIB = -DZLIB_API will substitute the zlib API by libnxz's, removing that and you will have libnxz standalone and can have both libraries.

Reply:
@rzinsly Why can't we have both libraries providing the same symbols?

Reply:
@tuliom: If we distribute both libraries I guess we should be able to use both, right?
I recognize that it would not be a common usage anyway.

Reply:
@rzinsly Yes.  We won't stop distributing zlib.  We need to have both.
AFAIU, most users will not be interested on libnxz in the beginning, so we should not install it automatically.
But when they become interested, they have to be able to easily replace zlib with libnxz without having to recompile their software.  Assuming it has already been built with libnxz.

Reply:
> they have to be able to easily replace zlib with libnxz without having to recompile their software. Assuming it has already been built with libnxz.

I didn't understand this bit, to build with libnxz they would have to recompile the software, right? They can just use LD_PRELOAD but that is not the easiest way IMO.
I guess you mean that the user will not want to change the code to start using the nx_ symbols? It makes sense, I'm ok with keeping -DZLIB_API if that is the case. But I don't see a problem on having it providing only the nx_ symbols for a full standalone library, also it could be used in the unlikely scenario that someone wants to use the NX accelerator for some things and zlib for others in the same application.
I'm ok with whatever the team decides it's the behaviour the library should have on AT.

Reply:
@rzinsly I'm sorry. I meant to say: "Assuming it has already been built with AT".
If they have built their software with AT, they can use `LD_PRELOAD` to replace zlib with libnxz.
Which way do you think is easiest for customers?

> I guess you mean that the user will not want to change the code to start using the nx_ symbols?

Exactly.

> in the unlikely scenario that someone wants to use the NX accelerator for some things and zlib for others in the same application.

In order to do this, this user would have to change their code, regardless of which symbols are exported in libnxz.
IMHO, it's technically possible to do this even when libnxz is compiled with `-DZLIB_API` provided that upstream libnxz is prepared for this. If it isn't we can always submit bug reports upstream.

Reply:
> If they have built their software with AT, they can use `LD_PRELOAD` to replace zlib with libnxz.
> Which way do you think is easiest for customers?

I didn't mean to say that use LD_PRELOAD is difficult, I just don't think that exporting or keep setting it is not easier than recompile the program once even for big applications.

Reply:
So I will keep the -DZLIB_API.